src/extension.ts

```diff
@@ -14,111 +11,13 @@ import {
   LanguageClient,
   LanguageClientOptions,
   RequestType,
-  StreamInfo,
   TextDocumentIdentifier,
 } from "vscode-languageclient";
 import { getCoursierExecutable } from "./coursier/coursier";
 
 let client: LanguageClient;
 
 export function activate(context: ExtensionContext) {
-  function createServer(): Promise<StreamInfo> {
-    function startServer(executable: string): Promise<StreamInfo> {
-      console.log(`Executable located at ${executable}.`);
-      return new Promise((resolve, reject) => {
-        const server = net
-          .createServer((socket) => {
-            console.log("Creating server");
-
-            resolve({
-              reader: socket,
-              writer: socket,
-            });
-
-            socket.on("end", () => console.log("Disconnected"));
-          })
-          .on("error", (err) => {
-            // handle errors here
-            reject(err);
-          });
-
-        // grab a random port.
-        server.listen(() => {
-          // Start the child java process
-          let options = { cwd: context.extensionPath };
-
-          let port = (server.address() as net.AddressInfo).port;
-
-          let version = vscode.workspace
-            .getConfiguration("smithyLsp")
-            .get("version", "`");
-
-          // Downloading latest poms
-          let resolveArgs = [
-            "resolve",
-            "--mode",
-            "force",
-            "com.disneystreaming.smithy:smithy-language-server:" + version,
-            "-r",
-            "m2local",
-          ];
-          let resolveProcess = child_process.spawn(
-            executable,
-            resolveArgs,
-            options
-          );
```

Comment:
Don't you want to leave this part in?

Reply:
there is no requirement to resolve first, if you have not, `launch` needs to resolve before it runs java

Reply:
right, I assumed the `force` part made sure you're on the latest version and we wanted to keep that

Reply:
@kubukoz is correct on this. That being said I'm not finding the documentation for the `--mode` option :/